crates/iceberg/src/spec/values.rs

```diff
@@ -70,6 +79,571 @@ pub enum PrimitiveLiteral {
+    ///
+    /// See [`NaiveTime::from_str`] for details.
+    ///
+    /// Example:
+    /// ```rust
+    /// use iceberg::spec::{Literal, Datum};
+    /// let t = Datum::time_from_str("01:02:01.888999777").unwrap();
+    ///
+    /// assert_eq!(&format!("{t}"), "01:02:01.888999");
+    /// ```
+    pub fn time_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
+        let t = s.as_ref().parse::<NaiveTime>().map_err(|e| {
+            Error::new(
+                ErrorKind::DataInvalid,
+                format!("Can't parse time from string: {}", s.as_ref()),
+            )
+            .with_source(e)
+        })?;
+
+        Ok(Self::time_from_naive_time(t))
+    }
+
+    /// Creates time literal from hour, minute, second, and microseconds.
+    ///
+    /// See [`NaiveTime::from_hms_micro_opt`].
+    ///
+    /// Example:
+    /// ```rust
+    ///
+    /// use iceberg::spec::{Literal, Datum};
+    /// let t = Datum::time_from_hms_micro(22, 15, 33, 111).unwrap();
+    ///
+    /// assert_eq!(&format!("{t}"), "22:15:33.000111");
+    /// ```
+    pub fn time_from_hms_micro(hour: u32, min: u32, sec: u32, micro: u32) -> Result<Self> {
+        let t = NaiveTime::from_hms_micro_opt(hour, min, sec, micro)
+            .ok_or_else(|| Error::new(
+                ErrorKind::DataInvalid,
+                format!("Can't create time from hour: {hour}, min: {min}, second: {sec}, microsecond: {micro}"),
+            ))?;
+        Ok(Self::time_from_naive_time(t))
+    }
+
+    /// Creates a timestamp from unix epoch in microseconds.
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    ///
+    /// use iceberg::spec::Datum;
+    /// let t = Datum::timestamp(1000);
+    ///
+    /// assert_eq!(&format!("{t}"), "1970-01-01 00:00:00.001");
+    /// ```
+    pub fn timestamp(value: i64) -> Self {
+        Self {
+            r#type: PrimitiveType::Timestamp,
+            literal: PrimitiveLiteral::Timestamp(value),
+        }
+    }
+
+    /// Creates a timestamp from [`DateTime`].
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    ///
+    /// use chrono::{TimeZone, Utc};
+    /// use iceberg::spec::Datum;
+    /// let t = Datum::timestamp_from_datetime(Utc.timestamp_opt(1000, 0).unwrap());
+    ///
+    /// assert_eq!(&format!("{t}"), "1970-01-01 00:16:40");
+    /// ```
+    pub fn timestamp_from_datetime<T: TimeZone>(dt: DateTime<T>) -> Self {
+        Self::timestamp(dt.with_timezone(&Utc).timestamp_micros())
+    }
+
+    /// Parse a timestamp in RFC3339 format.
+    ///
+    /// See [`DateTime<Utc>::from_str`].
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    /// use chrono::{DateTime, FixedOffset, NaiveDate, NaiveDateTime, NaiveTime};
+    /// use iceberg::spec::{Literal, Datum};
+    /// let t = Datum::timestamp_from_str("2012-12-12 12:12:12.8899-04:00").unwrap();
+    ///
+    /// assert_eq!(&format!("{t}"), "2012-12-12 16:12:12.889900");
+    /// ```
+    pub fn timestamp_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
```

Comment:
I think this one has a timezone (at least in the example in the comments):
```suggestion
    pub fn timestamptz_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
```
Another option is to detect if there is a timezone postfix (`-4:00`) and based on that return a `timestamp` or `timestamptz`

Reply:
> Another option is to detect if there is a timezone postfix (-4:00) and based on that return a timestamp or timestamptz

Good suggest, I think we should provide another api for this option, while keeping the original since it's also useful.

Reply:
After checking [`DateTime` api](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#impl-FromStr-for-DateTime%3CFixedOffset%3E), it only accepts [rfc3339](https://datatracker.ietf.org/doc/html/rfc3339), which  must contains a offset. Given that it's not trivial to parse datetime format manually, I would prefer to add the extra api when necessary. What do you think?

Reply:
Since timestamps cause a lot of confusion, it is important to fix this properly in the APIs. Would this only accept `Utc` timezones? It would also be good just to add some tests here. For example: https://github.com/apache/iceberg-python/blob/6708a6eaa76a2b4aab58601f10210f778b3f03a4/tests/expressions/test_literals.py#L326-L340

Reply:
I realise that for timestamp without tz, we should parse it to `NaiveDateTime`, and now everything works. The `timestamp_from_str` can't accept string with timezone value, and `timestamptz_from_str` 's input must contains a timezone. Please see tests here:
https://github.com/liurenjie1024/iceberg-rust/blob/766d77ae807e464f703093b914dfada45274ed9e/crates/iceberg/src/spec/values.rs#L2828
https://github.com/liurenjie1024/iceberg-rust/blob/766d77ae807e464f703093b914dfada45274ed9e/crates/iceberg/src/spec/values.rs#L2843